### ISSUE TITLE
Stream read query

### ISSUE THREAD

**Issue by sonic182** [balance-platform/pillar#39]
The lib doesn't support read response as stream isn't it?

**Comment by sofakingworld** [comment]
Yes, lib doesn't support this.

Is this supported via HTTP?

**Comment by sonic182** [comment]
Yes, when the server respond with headers `Transfer-Encoding: chunked` and then, the body is <chunk_size><chunk><chunk_size><chunk>....  See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding#examples

elixir Finch lib support this, with `Finch.request/6` method. I have use Finch because I needed to download clickhouse response in stream

**Comment by sonic182** [comment]
You can simplify this package A LOT by just using Finch, which already provides pool of conns

**Comment by sofakingworld** [comment]
Right now we paused development of Pillar until September, in this way i advise you to make fork with branch https://github.com/balance-platform/pillar/pull/26 and add Finch HTTP Client

**Comment by hkrutzer** [comment]
With a stream if you need to query a lot of data to do processing on, you can do so without buffering everything first. I also need this and I think @sonic182 is right that Finch seems like a great option, as it does pooling out of the box and also supports streaming.
